src/main/java/com/exactpro/th2/taskutils/FutureTracker.java

```diff
@@ -98,27 +209,81 @@ public void awaitRemaining (long timeoutMillis) {
                     }
                 }
             } catch (ExecutionException | TimeoutException e) {
+                // do nothing
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
```

Comment:
note: We probably should break the loop in this case - no reason to continue since the thread is interrupted and every call to `future.get` might throw `InterruptedException`